docs/Generators.md

```diff
@@ -209,6 +209,36 @@ Whenever a new Helm chart/Kustomize YAML/Application/plain subfolder is added to
 
 As with other generators, clusters *must* already be defined within Argo CD, in order to generate Applications for them.
 
+The generator also supports `exclude` option in order to exclude directories in the repository from being created by the applicationset;
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: ApplicationSet
+metadata:
+  name: cluster-addons
+spec:
+  generators:
+  - git:
+      repoURL: https://github.com/argoproj-labs/applicationset.git
+      revision: HEAD
+      directories:
+      - path: examples/git-generator-directory/cluster-addons/*
+      - exclude: true
```

Comment:
I am not sure what this yaml for, is it just a copy of this [example](https://github.com/argoproj-labs/applicationset/blame/7717fd6870e11cac48dde8ea39a3784bad8cfe80/docs/Generators.md#L177-L199) in Generators.md ?
If so , should we add to the same yaml or to add a new `/examples/git-generator-directory/git-directories-exclude-example.yaml` ?

Reply:
Great questions, how about this:

My goal is to keep the original example the same (eg I would like the original example to only create apps for the 2 existing cluster addons), and introduce your change as a new example. 

So let's split your change into a new example, under the `examples` directory, something like this:

1) Create a new excludes directory, under `git-generator-directory`, eg `(repo root)/examples/git-generator-directory/excludes/`

2) Copy the `cluster-addons` directory in your PR into this new directory
  - It will be eg `(repo root)/examples/git-generator-directory/excludes/cluster-addons`, and this directory will contain `exclude-helm-guestbook` plus the two existings addons.

3) Create a new example yaml `(repo root)/examples/git-generator-directory/excludes/git-directories-exclude-example.yaml` containing the exclude example you've got in the doc, and have it point to the above directory.

4) Finally, update the example in the doc to point to this new example in the above `examples` directory.

Overall, this means the two examples can be kept separate, and each can demonstrate different functionality.

Reply:
done